sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/Validations.scala

```diff
@@ -0,0 +1,362 @@
+    noSubscriptionMethodWithAcl(component) ++
+    noSubscriptionWithRestAnnotations(component) ++
+    subscriptionMethodMustHaveOneParameter(component)
+  }
+
+  def validate(component: Class[_]): Validation =
+    validateAction(component) ++
+    validateView(component)
+
+  private def validateAction(component: Class[_]): Validation = {
+    when[Action](component) {
+      commonValidation(component) ++ commonSubscriptionValidation(component)
+    }
+  }
+
+  private def validateView(component: Class[_]): Validation = {
+    when[View[_]](component) {
+      commonValidation(component) ++
+      commonSubscriptionValidation(component) ++
+      viewMustHaveTableName(component) ++
+      viewMustHaveOneQueryMethod(component) ++
+      streamUpdatesQueryMustReturnFlux(component)
+    }
+  }
+
+  private def errorMessage(element: AnnotatedElement, message: String): String = {
+    val elementStr =
+      element match {
+        case clz: Class[_] => clz.getName
+        case meth: Method  => s"${meth.getDeclaringClass.getName}#${meth.getName}"
+        case any           => any.toString
+      }
+    s"On '$elementStr': $message"
+  }
+
+  private def eventSourcedEntitySubscriptionValidations(component: Class[_]): Validation = {
+    val methods = component.getMethods.toIndexedSeq
+    when(
+      hasEventSourcedEntitySubscription(component) &&
+      methods.exists(hasEventSourcedEntitySubscription)) {
+      // collect offending methods
+      methods.filter(hasEventSourcedEntitySubscription).map { method =>
+        errorMessage(
+          method,
+          "You cannot use @Subscribe.EventSourcedEntity annotation in both methods and class. You can do either one or the other.")
+      }
+    }
+  }
+
+  private def topicSubscriptionValidations(component: Class[_]): Validation = {
+    val methods = component.getMethods.toIndexedSeq
+    when(hasTopicSubscription(component) && methods.exists(hasTopicSubscription)) {
+      // collect offending methods
+      methods.filter(hasTopicSubscription).map { method =>
+        errorMessage(
+          method,
+          "You cannot use @Subscribe.Topic annotation in both methods and class. You can do either one or the other.")
+      }
+    }
+  }
+
+  private def publishStreamIdMustBeFilled(component: Class[_]): Validation = {
+    Option(component.getAnnotation(classOf[Publish.Stream]))
+      .map { ann =>
+        when(ann.id().trim.isEmpty) {
+          Seq("Publish.Stream id can not be an empty string")
+        }
+      }
+      .getOrElse(Valid)
+  }
+
+  private def noSubscriptionMethodWithAcl(component: Class[_]): Validation = {
+
+    val hasSubscriptionAndAcl = (method: Method) => hasAcl(method) && hasSubscription(method)
+
+    val messages =
+      component.getMethods.toIndexedSeq.filter(hasSubscriptionAndAcl).map { method =>
+        errorMessage(
+          method,
+          "Subscription methods are for internal use only and cannot be combined with ACL annotations.")
```

Comment:
```suggestion
          "Methods annotated with Kalix @Subscription annotations are for internal use only and cannot be annotated with ACL annotations.")
```
For consistency with the message below